core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -281,6 +284,73 @@ private SdkClientConfiguration mergeGlobalDefaults(SdkClientConfiguration config
                                          .option(CRC32_FROM_COMPRESSED_DATA_ENABLED, false));
     }
 
+    private SdkClientConfiguration addCompressionConfigGlobalDefaults(SdkClientConfiguration configuration,
+                                                                      Supplier<ProfileFile> profileFileSupplier) {
+        Optional<Boolean> requestCompressionEnabled = getCompressionEnabled(configuration);
+        Optional<Integer> minCompressionThreshold = getCompressionThreshold(configuration);
+
+        if (requestCompressionEnabled.isPresent() && minCompressionThreshold.isPresent()) {
+            return configuration;
+        }
+
+        Boolean compressionEnabled = requestCompressionEnabled.orElse(null);
+        Integer compressionThreshold = minCompressionThreshold.orElse(null);
+
+        if (compressionEnabled == null) {
+            Optional<Boolean> systemSetting = SdkSystemSetting.AWS_DISABLE_REQUEST_COMPRESSION.getBooleanValue();
+            if (systemSetting.isPresent()) {
+                compressionEnabled = !systemSetting.get();
+            } else {
+                String profileName = ProfileFileSystemSetting.AWS_PROFILE.getStringValueOrThrow();
```

Comment:
We should use the PROFILE_NAME from the SdkClientConfiguration, not from environment variables. These could differ, if the customer has set profileName on ClientOverrideConfiguration.